Content.Shared/GameObjects/Components/Mobs/SharedStatusEffectsComponent.cs

```diff
@@ -18,14 +18,22 @@ public abstract class SharedStatusEffectsComponent : Component
     [Serializable, NetSerializable]
     public class StatusEffectComponentState : ComponentState
     {
-        public Dictionary<StatusEffect, string> StatusEffects;
+        public Dictionary<StatusEffect, StatusEffectStatus> StatusEffects;
 
-        public StatusEffectComponentState(Dictionary<StatusEffect, string> statusEffects) : base(ContentNetIDs.STATUSEFFECTS)
+        public StatusEffectComponentState(Dictionary<StatusEffect, StatusEffectStatus> statusEffects) : base(ContentNetIDs.STATUSEFFECTS)
         {
             StatusEffects = statusEffects;
         }
     }
 
+    [Serializable, NetSerializable]
+    public struct StatusEffectStatus
+    {
+        public string Icon;
+        public TimeSpan? CooldownStart;
```

Comment:
You could do `(TimeSpan CooldownStart, TimeSpan CooldownEnd)?`, right? Since they're always going to be set at the same time.

Reply:
Indeed, you're right. Changed now.